src/components/sidebar/routePathView/NewRoutePathView.tsx

```diff
@@ -4,98 +4,111 @@ import classnames from 'classnames';
-        const newRoutePath = RoutePathFactory.createNewRoutePath('1', '1');
-        this.props.routePathStore!.setRoutePath(newRoutePath);
+    public onChange = (property: string) => (value: string) => {
+        this.setState({
+            currentRoutePath: { ...this.state.currentRoutePath!, [property]: value },
+        });
     }
 
-    // TODO: Use routePath store instead of state
-    // TODO, this is just a placeholder, implement real logic for creating new routePaths
-    private getInitialRoutePath = () => {
-        const newRoutePath: IRoutePath = {
-            internalId: '',
-            routeId: '',
-            routePathName: 'Uusi reitinsuunta',
-            routePathNameSw: 'Ny ruttriktning',
-            direction: '1',
-            positions: [[0, 0]],
-            geoJson: null,
-            visible: true,
-            startTime: new Date,
-            endTime: new Date,
-            lastModified: new Date,
-            routePathLinks: [],
-            originFi: '',
-            originSw: '',
-            destinationFi: '',
-            destinationSw: '',
-            routePathShortName: '',
-            routePathShortNameSw: '',
-            modifiedBy: '',
-        };
-        return newRoutePath;
+    public onSave = () => {
+        // TODO
     }
 
-    // TODO
-    public onChange = () => {
+    componentWillMount() {
+        if (!this.props.routePathStore!.routePath) {
+            this.initEmptyRoutePath();
+        }
     }
 
-    // TODO
-    public onSave = () => {
+    initEmptyRoutePath() {
+        const queryParams = navigator.getQueryParamValues();
+        const newRoutePath =
+            RoutePathFactory.createNewRoutePath(queryParams.lineId, queryParams.routeId);
 
+        this.props.routePathStore!.setRoutePath(newRoutePath);
     }
 
     componentDidMount() {
-        this.props.mapStore!.setNodeSize(NodeSize.large);
+        this.initStores();
+        this.initCurrentRoutePath();
+    }
 
+    private initStores() {
+        this.props.mapStore!.setNodeSize(NodeSize.large);
         this.props.networkStore!.setIsNodesVisible(true);
         this.props.networkStore!.setIsLinksVisible(true);
-
         this.props.routePathStore!.setIsCreating(true);
     }
 
+    private initCurrentRoutePath() {
+        // TODO: this should be action call
+        this.props.routeStore!.routes = [];
+
+        if (!this.state.currentRoutePath) {
+            const currentRoutePath = this.props.routePathStore!.routePath;
+            this.setState({
+                currentRoutePath:  currentRoutePath ? currentRoutePath : undefined,
+            });
+        }
+    }
+
     componentWillUnmount() {
         this.props.mapStore!.setNodeSize(NodeSize.normal);
         this.props.routePathStore!.setIsCreating(false);
-
+        this.props.routePathStore!.setRoutePath(null);
     }
 
     public render(): any {
+        const currentRoutePath = this.state.currentRoutePath;
+        if (!currentRoutePath) {
```

Comment:
This sould probably never happend if we do this right

Reply:
I assume this is unnecessary if we move the initiation to the constructor.

Reply:
If I remove this if, then it says in render: "Object possibly undefined". So returning null if currentRoutePath is falsey should be fine

Reply:
Does it need to be optional? I mean couldn't the ?-mark just be removed so it's always defined?